vm/src/stdlib/ast.rs

```diff
@@ -349,321 +258,254 @@ fn expression_to_ast(vm: &VirtualMachine, expression: &ast::Expression) -> PyObj
+                }
+                ast::ComprehensionKind::Dict { .. } => {
+                    node!(vm, DictComp, {generators => py_generators})
+                }
+            }
         }
         ast::Expression::Yield { value } => {
-            let node = create_node(vm, "Yield");
-
             let py_value = match value {
-                Some(value) => expression_to_ast(vm, value),
+                Some(value) => expression_to_ast(vm, value)?.into_object(),
                 None => vm.ctx.none(),
             };
-            vm.set_attr(&node, "value", py_value).unwrap();
-
-            node
+            node!(vm, Yield, {
+                value => py_value
+            })
         }
         ast::Expression::YieldFrom { value } => {
-            let node = create_node(vm, "YieldFrom");
-
-            let py_value = expression_to_ast(vm, value);
-            vm.set_attr(&node, "value", py_value).unwrap();
-
-            node
-        }
-        ast::Expression::Subscript { a, b } => {
-            let node = create_node(vm, "Subscript");
-
-            let py_value = expression_to_ast(vm, a);
-            vm.set_attr(&node, "value", py_value).unwrap();
-
-            let py_slice = expression_to_ast(vm, b);
-            vm.set_attr(&node, "slice", py_slice).unwrap();
-
-            node
-        }
-        ast::Expression::Attribute { value, name } => {
-            let node = create_node(vm, "Attribute");
-
-            let py_value = expression_to_ast(vm, value);
-            vm.set_attr(&node, "value", py_value).unwrap();
-
-            let py_attr = vm.ctx.new_str(name.to_string());
-            vm.set_attr(&node, "attr", py_attr).unwrap();
-
-            node
-        }
-        ast::Expression::Starred { value } => {
-            let node = create_node(vm, "Starred");
-
-            let py_value = expression_to_ast(vm, value);
-            vm.set_attr(&node, "value", py_value).unwrap();
-
-            node
-        }
-        ast::Expression::Slice { elements } => {
-            let node = create_node(vm, "Slice");
-
-            let py_value = expressions_to_ast(vm, elements);
-            vm.set_attr(&node, "bounds", py_value).unwrap();
-
-            node
-        }
+            let py_value = expression_to_ast(vm, value)?;
+            node!(vm, YieldFrom, {
+                value => py_value
+            })
+        }
+        ast::Expression::Subscript { a, b } => node!(vm, Subscript, {
+            value => expression_to_ast(vm, a)?,
+            slice => expression_to_ast(vm, b)?,
+        }),
+        ast::Expression::Attribute { value, name } => node!(vm, Attribute, {
+            value => expression_to_ast(vm, value)?,
+            attr => vm.ctx.new_str(name.to_string()),
+        }),
+        ast::Expression::Starred { value } => node!(vm, Starred, {
+            value => expression_to_ast(vm, value)?
+        }),
+        ast::Expression::Slice { elements } => node!(vm, Starred, {
```

Comment:
This should no be a Starred node, but a Slice node.

Reply:
Well spotted!